### DISCUSSION TITLE
poolsize does not match sum of drive sizes at all (not even close)

### DISCUSSION THREAD

**Discussion by ilkerol** [trapexit/mergerfs#1340] [Q&A]
Maybe I am somehow stupid, but I added 5 drives to one mergerfs pool (OMV latest version, i attached a screenshot):
3x 16,7 TB  
2x 2,7 TB
(sda1, sdb1, sdc1, sdd1, sdf1)

I did not setup any parity stuff and no snapraid and so on yet. Just one pool with those drives, mounted to /srv/mergerfs/data. The weird thing is, that it shows up with 22TB capacity, not 55TB as it should be... all drives are formatted, each individual drive shows up with correct size when i check df -h. But the pool size is off by 33TB. 

Does anyone have a clue what is going on here? 
![omv file system](https://github.com/trapexit/mergerfs/assets/156719153/4eeadd07-f4de-4526-b325-6d8c39efbe03)

**Comment by trapexit** [comment]
Is that OMV? I don't own or maintain such 3rd party software so I can only speak to core mergerfs setups. If you wish to share your actual config and all the details I request in the support section of the docs then I might be able to help.

**Reply by ilkerol** [reply]
Yes, that is omv, but ill get you the config and details you request, will take a while. Thanks for the fast reply!

**Reply by ilkerol** [reply]
Ok, here is the requested information:

- Broader problem see above (sum of drive sizes 55TB does not match pool size of 22,8TB).

- Attempted solutions: rebuild the pool, take out a drive and put it back in, check "df -h" if actual drive sizes are recognized correctly by the system.

- mergerfs v2.39.0

- mergerfs settings from OMV plugin:
name of the pool "data", file systems added to pool sda1, sdb1, sdc1, sdd1, sdf1 (all ext4 and individually mounted to /srv), did not use the options to convert filesystems to paths or shared folders to paths. create policy "existing path - most free space", minimum free space 4 GB, options are "defaults, cache.files=off"
**maybe the output of ps -eaf is helpful:**
_mergerfs data:abc9bcc5-9f65-4327-980b-2d764d657c58 /srv/mergerfs/data -o rw,branches=/srv/dev-disk-by-uuid-61930138-cd9b-4af6-9c8c-1a16c1d10bc3:/srv/dev-disk-by-uuid-ccfbfeb7-c0fc-4443-bcb8-c88d01b98fc3:/srv/dev-disk-by-uuid-ed92cdaf-cdec-4690-a638-9f03b7ddb1b0,category.create=epmfs,minfreespace=4G,fsname=data:abc9bcc5-9f65-4327-980b-2d764d657c58,cache.files=off,dev,suid_

- OS: Linux OMV 6.1.0-21-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.90-1 (2024-05-03) x86_64 GNU/Linux
- lsb _release -a gives me:
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 12 (bookworm)
Release:        12
Codename:       bookworm

- df -h gives me the same as upper screenshot showed (excuse the german column names):
_Dateisystem                               Größe Benutzt Verf. Verw% Eingehängt auf
udev                                        16G       0   16G    0% /dev
tmpfs                                      3,2G    4,4M  3,2G    1% /run
/dev/nvme0n1p2                             1,8T    5,1G  1,7T    1% /
tmpfs                                       16G       0   16G    0% /dev/shm
tmpfs                                      5,0M       0  5,0M    0% /run/lock
tmpfs                                       16G       0   16G    0% /tmp
/dev/nvme0n1p1                             511M    148K  511M    1% /boot/efi
/dev/sdg1                                  938G    2,1M  938G    1% /srv/dev-disk-by-uuid-8e7135b1-6056-48a0-b39b-ce4825c16110
/dev/sdb1                                  2,7T    2,1M  2,7T    1% /srv/dev-disk-by-uuid-ed92cdaf-cdec-4690-a638-9f03b7ddb1b0
/dev/sdc1                                  2,7T    2,1M  2,7T    1% /srv/dev-disk-by-uuid-61930138-cd9b-4af6-9c8c-1a16c1d10bc3
/dev/sde1                                   17T    2,1M   17T    1% /srv/dev-disk-by-uuid-f4c48070-c202-4a95-80b2-9aff1b5c4ade
/dev/sda1                                   17T    1,3T   16T    8% /srv/dev-disk-by-uuid-ccfbfeb7-c0fc-4443-bcb8-c88d01b98fc3
data:abc9bcc5-9f65-4327-980b-2d764d657c58   22T    1,3T   21T    6% /srv/mergerfs/data
/dev/sdd1                                   17T    2,1M   17T    1% /srv/dev-disk-by-uuid-0e298d89-359d-4349-88e8-517a0ccc364a
/dev/sdf1                                   17T    2,1M   17T    1% /srv/dev-disk-by-uuid-0d1e0dd0-b514-4ceb-be92-0fa4e55c2d5a_

- the mergerfs pool is mounted to /srv/mergerfs/data
- no docker involved


I got a hunch... if you look at that ps i pasted it lists only 3 drives under "brances = " - and those add up to the listed 22.8TB. So it seems the omv plugin did not add the 2 additional 16,7TB drives, although according to the GUI it did. 
I am no expert, but then that seems to be an OMV GUI problem with the plugin? 
So I would have to somehow manually configure the plugin I assume?

**Reply by trapexit** [reply]
branches=/srv/dev-disk-by-uuid-61930138-cd9b-4af6-9c8c-1a16c1d10bc3:/srv/dev-disk-by-uuid-ccfbfeb7-c0fc-4443-bcb8-c88d01b98fc3:/srv/dev-disk-by-uuid-ed92cdaf-cdec-4690-a638-9f03b7ddb1b0

I see 3 branches. bc3 = 2.7T, fc3 = 17T, 1b0 = 2.7T

2.7 + 17 + 2.7 = 22.4

**Reply by trapexit** [reply]
I've never used OMV with mergerfs plugin. You'll need to talk to them about it if it isn't working as expected.

**Reply by ilkerol** [reply]
ok, then the issue really is that the added drives via the GUI do not really end up in the pool, although the omv gui clearly shows them as being part of it:
data
/srv/dev-disk-by-uuid-61930138-cd9b-4af6-9c8c-1a16c1d10bc3
/srv/dev-disk-by-uuid-ccfbfeb7-c0fc-4443-bcb8-c88d01b98fc3
/srv/dev-disk-by-uuid-ed92cdaf-cdec-4690-a638-9f03b7ddb1b0
/srv/dev-disk-by-uuid-0e298d89-359d-4349-88e8-517a0ccc364a
/srv/dev-disk-by-uuid-0d1e0dd0-b514-4ceb-be92-0fa4e55c2d5a

Think I will just manually add them to the mergerfs config... thank you a lot, just researching the required information made the problem obvious :)

**Reply by trapexit** [reply]
Did you restart after adding them? Did you check /etc/fstab or wherever it stores this info? The plugin doesn't change things live. While mergerfs does have a runtime API most every filesystem ever requires remounting for settings to change.

**Reply by ilkerol** [reply]
Maybe that was the issue... the omv frontend asks you to "apply pending changes" after everything you do, so I just assumed it would really apply changes. But apparently it didnt. I restarted the server and now it works. So their "apply pending changes" is not to be trusted. Good to know!

**Reply by trapexit** [reply]
It applies the changes to the config. But that doesn't mean the thing picks up those changes. If you change settings to some other mounted filesystem it won't remount the filesystem. Might be the same for random other things in OMV. Would be nice if it indicated the difference between config and runtime.

**Comment by ryecoaaron** [comment]
Please open a thread on the forum or an issue on github.
https://forum.openmediavault.org/
https://github.com/OpenMediaVault-Plugin-Developers/openmediavault-mergerfs/issues
